src/main/java/calculator/domain/preprocess/DelimiterSegmentProcessor.java

```diff
@@ -0,0 +1,39 @@
+package calculator.domain.preprocess;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static calculator.utils.Util.isInteger;
+
+public class DelimiterSegmentProcessor {
+
+    public boolean checkCustomDelimiterRequest(String input) {
+        if (input.startsWith("//") && input.contains("\n")) {
+            validateCustomDelimiterRequest(input);
+            return true;
+        } else
+            return false;
+    }
+
+    //구분자가 두 글자 이상일 경우 가령 ",," 와 같은 것을 구분자로 받는 경우 기존의 구분자와 충돌이 발생할 수 있다고 판단하여 제한.
+    private void validateCustomDelimiterRequest(String input) {
+        if (input.indexOf("\n") == 2)
+            throw new IllegalArgumentException("구분자로 사용할 문자 미입력");
+        if (input.indexOf("\n") > 3)
+            throw new IllegalArgumentException("구분자로 사용할 문자가 두 글자 이상");
+        if (isInteger(Character.toString(input.charAt(2))))
+            throw new IllegalArgumentException("구분자로 숫자를 사용할 수 없습니다");
+    }
+
+    public String extractCalculationSegment(String input) {
+        return input.substring(4);
+    }
+
+    public List<String> extractDelimiterList(String input) {
+        String delimiter = Character.toString(input.charAt(2));
+        List<String> delimiterList = new ArrayList<>(Arrays.asList(",", ":"));
```

Comment:
```suggestion
        List<String> delimiterList = List.of(",",":");
```

이렇게 하면 더 적은 코드를 작성하실 수 있을거에요 언제나왔는지 정확히 기억은 안나는데 Java7인가 8부터 이렇게 사용가능하도록 바뀐걸로 알고있어요 !

Reply:
저도 처음에는 저렇게 만들었는데 불변 리스트가 만들어져 add가 안되는 부분 때문에 수정했습니다! 기억해두시면 좋을 것 같아요 😄